### PR TITLE
GH-153 - Create facades for CartModule

### LINKED ISSUE #153: Create facades for CartModule
Follow instructions from #6 and see example/spike in #140.
If you have a question, post in team slack with @tobias and @wei.

### PR DESCRIPTION
**Pull request by znikola**
Closes #153 

Note for the reviewer: the `cart.service.spec.ts` is `xdescribe`d, as it required some refactoring, so please ignore it for now.

### CODE REVIEW

File: projects/storefrontlib/src/lib/cart/components/add-to-cart/add-to-cart.component.ts [hunk 1]
```diff
@@ -34,16 +35,13 @@ export class AddToCartComponent implements OnInit {
...
+    private modalService: NgbModal
   ) {}
 
   ngOnInit() {
     if (this.productCode) {
-      this.loaded$ = this.store.pipe(select(fromCartStore.getLoaded));
-      this.cartEntry$ = this.store.pipe(
-        select(fromCartStore.getEntrySelectorFactory(this.productCode))
-      );
+      this.loaded$ = this.cartService.loaded$;
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/lib/cart/components/add-to-cart/add-to-cart.component.ts]: this 'loaded$' variable seems not used.

File: projects/storefrontlib/src/lib/cart/components/add-to-cart/add-to-cart.component.spec.ts [hunk 2]
```diff
@@ -26,6 +21,12 @@ class MockCartService {
       '1234': { entryNumber: 0, product: { code: productCode } }
     });
   }
+  getLoaded(): Observable<boolean> {
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/lib/cart/components/add-to-cart/add-to-cart.component.spec.ts]: it's better to change it to the property.

File: projects/storefrontlib/src/lib/cart/facade/cart.service.ts [hunk 3]
```diff
@@ -1,18 +1,21 @@
...
 import * as fromSelector from '../store/selectors';
-
-import * as fromAuthSelectors from './../../auth/store/selectors';
+import { AuthService } from '../../auth/index';
 
 import { ANONYMOUS_USERID, CartDataService } from './cart-data.service';
 
 @Injectable()
 export class CartService {
+  callback: Function;
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/lib/cart/facade/cart.service.ts]: this one should be 'private' or 'protected', right?

File: projects/storefrontlib/src/lib/cart/components/add-to-cart/add-to-cart.component.spec.ts (hunk 2 above)
Comment (znikola) [reply projects/storefrontlib/src/lib/cart/components/add-to-cart/add-to-cart.component.spec.ts]: Done.

File: projects/storefrontlib/src/lib/cart/facade/cart.service.ts (hunk 3 above)
Comment (znikola) [reply projects/storefrontlib/src/lib/cart/facade/cart.service.ts]: Done.

**Pull request merged**